test/extensions/filters/http/grpc_json_reverse_transcoder/utils_test.cc

```diff
@@ -0,0 +1,172 @@
+                          "shelf.editions=kindle&shelf.editions=hardback&shelf.editions"
+                          "=audiobook");
+
+  query_string.clear();
+  BuildQueryParamString(object, {}, &query_string, "unit_test");
+  EXPECT_EQ(query_string, "unit_test.books=book1&unit_test.books=book2&unit_test.description=This%"
+                          "20is%20a%20test%20description&unit_test.parent=projects%2F123456789&"
+                          "unit_test.publisher.isbn=slkdjfweoiru234&unit_test.publisher.name=pub1&"
+                          "unit_test.publisher.isbn=lksdjflkj3423&unit_test.publisher.name=pub2&"
+                          "unit_test.shelf.active=true&unit_test.shelf.authors.name=author1&unit_"
+                          "test.shelf.authors.name=author2&unit_test.shelf.code=3&unit_test.shelf."
+                          "content=Some%20random%20content&unit_test.shelf.editions=kindle&unit_"
+                          "test.shelf.editions=hardback&unit_test.shelf.editions=audiobook&unit_"
+                          "test.shelf.name=science-fiction&unit_test.theme=Kids");
+}
+
+TEST(BuildPath, ReturnsPathWithPlaceholdersReplaced) {
+  nlohmann::json request = R"({
+"shelf": {
+  "name": "science-fiction",
+  "code": 3,
+  "content": "Some random content",
+  "active": true,
+  "editions": ["kindle", "hardback", "audiobook"],
+  "authors": [
+    {"name": "author1"},
+    {"name": "author2"}
+  ]
+},
+"books": ["book1", "book2"],
+"parent": "projects/123456789",
+"theme": "Kids",
+"description": "This is a test description"
+})"_json;
+
+  // Check just the path whole message the request body.
+  absl::StatusOr<std::string> path =
+      BuildPath(request, "/v1/{parent=projects/*}/shelves/{shelf.name}", "*");
+  ASSERT_TRUE(path.ok());
+  EXPECT_EQ(path.value(), "/v1/projects/123456789/shelves/science-fiction");
+
+  // Check the path with a single field as the request body.
+  path = BuildPath(request, "/v1/{parent=projects/*}/shelves/{shelf.name}", "shelf");
+  ASSERT_TRUE(path.ok());
+  EXPECT_EQ(path.value(), "/v1/projects/123456789/shelves/"
+                          "science-fiction?books=book1&books=book2&description=This%20is%20a%"
+                          "20test%20description&theme=Kids");
+
+  // Check path without any body fields.
+  path = BuildPath(request, "/v1/{parent=projects/*}/shelves/{shelf.name}", "");
+  ASSERT_TRUE(path.ok());
+  EXPECT_EQ(path.value(), "/v1/projects/123456789/shelves/"
+                          "science-fiction?books=book1&books=book2&"
+                          "description=This%20is%20a%20test%20description&"
+                          "shelf.active=true&shelf.authors.name=author1&"
+                          "shelf.authors.name=author2&"
+                          "shelf.code=3&shelf.content=Some%20random%20content&"
+                          "shelf.editions=kindle&shelf.editions=hardback&"
+                          "shelf.editions=audiobook&"
```

Comment:
struggling to understand why but this is triggering a spellcheck failure

Reply:
It's because of the `audobook`; changed it to `audio-book` to fix this.